common/src/main/java/com/microsoft/identity/common/internal/fido/LegacyFido2ApiManager.kt

```diff
@@ -0,0 +1,109 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.fido
+
+import android.content.Context
+import android.os.CancellationSignal
+import com.google.android.gms.fido.fido2.Fido2ApiClient
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredentialDescriptor
+import com.google.android.gms.tasks.OnSuccessListener
+import com.microsoft.identity.common.internal.providers.oauth2.WebViewAuthorizationFragment
+import com.microsoft.identity.common.java.opentelemetry.AttributeName
+import com.microsoft.identity.common.logging.Logger
+import io.opentelemetry.api.trace.Span
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.ArrayList
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * Utilizes the legacy Android FIDO2 API in order to return an attestation.
+ * Note that the legacy FIDO2 API should only be used for older Android versions which need it,
+ * such as Android 13 and below for FIDO2 security key support.
+ */
+class LegacyFido2ApiManager (val context: Context, val fragment: WebViewAuthorizationFragment) : IFidoManager {
+
+    val TAG = LegacyFido2ApiManager::class.simpleName.toString()
+
+    private val legacyApi = Fido2ApiClient(context)
+
+    /**
+     * Interacts with the FIDO credential provider and returns an assertion.
+     *
+     * @param challenge AuthFidoChallenge received from the server.
+     * @param relyingPartyIdentifier rpId received from the server.
+     * @param allowedCredentials List of allowed credentials to filter by.
+     * @param userVerificationPolicy User verification policy string.
+     * @param span OpenTelemetry span.
+     * @return assertion
+     */
+    override suspend fun authenticate(challenge: String,
+                                      relyingPartyIdentifier: String,
+                                      allowedCredentials: List<String>?,
+                                      userVerificationPolicy: String,
+                                      span: Span):
+            String = suspendCancellableCoroutine { continuation ->
```

Comment:
nit: please rearrange indentation of each block level for readability.

Reply:
Hopefully the change in this commit helps: cef6e97bf8ae3da4badd6b59bcbfed4e63358aba